### PR TITLE
add connection string

### PR DESCRIPTION
**Pull request by sowu880**
Will use connection string to create blob container if connection string is given in configuration.

### CODE REVIEW

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Common/Configurations/StorageConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+namespace Microsoft.Health.Fhir.Synapse.Common.Configurations
+{
+    public class StorageConfiguration
+    {
+        [JsonProperty("externalStorageConnectionString")]
+        public string ExternalStorageConnectionString { get; set; } = string.Empty;
+
+        [JsonProperty("internalStorageConnectionString")]
+        public string InternalStorageConnectionString { get; set; } = string.Empty;
```
Comment (QuanWanxx) [inline FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Common/Configurations/StorageConfiguration.cs:16]: A question, in what cases we use internal or external connection string?

Comment (QuanWanxx) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Common/Configurations/StorageConfiguration.cs:16]: Seems they can be distinguished by **TokenCredentialTypes** config, can we only add a **StorageConnectionString** here?

File: FhirToDataLake/test/Microsoft.Health.Fhir.Synapse.DataWriter.UnitTests/DataWriterTests.cs [hunk 2]
```diff
@@ -42,7 +42,7 @@ public async Task GivenAValidParameter_WhenWritingToBlob_CorrectContentShouldBeW
             var streamData = new StreamBatchData(stream, 1, TestResourceType);
             await dataWriter.WriteAsync(streamData, "mockJob", 0, 0, _testDate);
 
-            var containerClient = new AzureBlobContainerClientFactory(new DefaultTokenCredentialProvider(new NullLogger<DefaultTokenCredentialProvider>()), new NullLoggerFactory()).Create(LocalTestStorageUrl, TestContainerName);
+            var containerClient = new AzureBlobContainerClientFactory(new DefaultTokenCredentialProvider(new NullLogger<DefaultTokenCredentialProvider>()), Options.Create(new StorageConfiguration()), new NullLoggerFactory()).Create(LocalTestStorageUrl, TestContainerName);
```
Comment (QuanWanxx) [inline FhirToDataLake/test/Microsoft.Health.Fhir.Synapse.DataWriter.UnitTests/DataWriterTests.cs:45]: Maybe create a static **StorageConfiguration** for this test file.

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Common/Configurations/StorageConfiguration.cs (hunk 1 above)
Comment (sowu880) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Common/Configurations/StorageConfiguration.cs:16]: Internal Resources : internal storage for table and queue.
External Resources :  ACR, data lake, FHIR server.

Internal storage could use internal connection string and External data lake could use external connection string. These two strings may be different.

File: FhirToDataLake/test/Microsoft.Health.Fhir.Synapse.DataWriter.UnitTests/DataWriterTests.cs (hunk 2 above)
Comment (sowu880) [reply FhirToDataLake/test/Microsoft.Health.Fhir.Synapse.DataWriter.UnitTests/DataWriterTests.cs]: Thanks, fixed it.

**Review by tongwu-msft — APPROVED** [review]

**Pull request merged**